compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/viewinterop/UIKitInteropElementHolder.uikit.kt

```diff
@@ -147,4 +149,30 @@ internal abstract class UIKitInteropElementHolder<T : InteropView>(
     override fun changeInteropViewIndex(root: InteropViewGroup, index: Int) {
         root.insertSubview(view = group, atIndex = index.toLong())
     }
+
+    /**
+     * Check that [group] doesn't entirely clip a child view with a [cgRect]
+     */
+    private fun isVisible(cgRect: CValue<CGRect>): Boolean =
+        CGRectIsEmpty(
+            CGRectIntersection(cgRect, group.bounds)
```

Comment:
`CGRectIntersectsRect(cgRect, group.bounds)` ?

Reply:
I considered using it, but there is a subtle difference:
[CGRectIntersectsRect](https://developer.apple.com/documentation/coregraphics/1454747-cgrectintersectsrect?language=objc) returns `true` if it's a `null` rectangle (absence of intersection).

It means it could still be `empty` rectangle, with either dimension equal to 0, which is still `not visible`.
[CGRectIsEmpty](https://developer.apple.com/documentation/coregraphics/1454917-cgrectisempty) includes both `null` and `empty` and this is exactly what we need.

Reply:
Deleted the duplicating review comment below.

Reply:
Ok, understood